version: '3.9'
name: 'gse'

services:

  gse-db:
    image: mysql:8.0.22
    container_name: gse-db
    restart: always
    volumes:
      - ./dbdata:/var/lib/mysql
      - ./mysql.cnf:/etc/mysql/conf.d/mysql.cnf
    environment:
      MYSQL_ROOT_PASSWORD: 9ijn8uhb7ygv
      MYSQL_DATABASE: gse
      MYSQL_USER: gseadmin
      MYSQL_PASSWORD: Lugano2020
      TZ: UTC
    healthcheck:
      test: mysqladmin ping -h localhost
      start_period: 10s
      interval: 10s
      timeout: 5s
      retries: 5

  gse-fe:
    build:
      context: ../
      dockerfile: docker/Dockerfile.fe
    image: 'gse/frontend:latest'
    container_name: gse-fe
    volumes:
      - ./nginx/default.dev.conf:/etc/nginx/conf.d/default.conf
    ports: [ "7030:80" ]
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 5
    depends_on:
      gse-app:
        condition: service_started
    networks:
      - default


  gse-app:
    build:
      context: ../
      dockerfile: docker/Dockerfile.be
    image: 'gse/backend:latest'
    container_name: gse-app
    environment:
      WAIT_HOSTS: gse-db:3306
      SPRING_DATASOURCE_URL: jdbc:mysql://gse-db:3306/gse?serverTimezone=UTC&useLegacyDatetimeCode=false&useUnicode=yes&characterEncoding=UTF-8
      SPRING_FLYWAY_URL: jdbc:mysql://gse-db:3306/gse?serverTimezone=UTC&useLegacyDatetimeCode=false&useUnicode=yes&characterEncoding=UTF-8
      APP_CRAWL_ENABLED: 'false'
    ports: [ "48001:8080" ]
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 5
    depends_on:
      gse-db:
        condition: service_healthy
    networks:
      - default

networks:
  default:
    name: gse-network
